dashboard/test/controllers/datablock_storage_controller_test.rb

```diff
@@ -0,0 +1,713 @@
+require "test_helper"
+
+class DatablockStorageControllerTest < ActionDispatch::IntegrationTest
+  setup_all do
+    @student = create :student
+
+    project = create :project, owner: @student
+    project.project_type = 'applab'
+    project.save!
+    @channel_id = project.channel_id
+  end
+
+  setup do
+    sign_in @student
+  end
+
+  def _url(action)
+    return "/datablock_storage/#{@channel_id}/#{action}"
+  end
+
+  def create_bob_record
+    post _url(:create_record), params: {
```

Comment:
`post :create_record, params: { ... `

Reply:
I tried just removing `_url()` and I got tons of errors because the generated URLs looked like: `https://test-studio.code.org:443create_record`, whereas the routes.rb entry looks like: `resources :datablock_storage, path: '/datablock_storage/:channel_id/'`.

More detail here: https://github.com/code-dot-org/code-dot-org/pull/57330#issuecomment-2002355449

Any suggestions @bethanyaconnor ? Given the `:channel_id` in the route path, I suspect we'd have to modify every post/get/delete/etc call to include a channel_id somehow, at which point I'm probably using a _url() helper anyway... it just might look a little different internally?

Mostly I'm wondering whether this signals there's something deeply wrong with our routes.rb configuration.

Reply:
hm let me think on this one a little...